src/combinator.rs

```diff
@@ -57,6 +57,65 @@ impl <P: Parser> Iterator for Iter<P> {
 }
 
 #[derive(Clone)]
+pub struct ChoiceSlice<'a, P>(&'a [P])
+    where P: Parser + 'a;
+
+impl <'a, I, O, P> Parser for ChoiceSlice<'a, P>
+    where I: Stream
+        , P: Parser<Input=I, Output=O> + Clone +  'a {
+    type Input = I;
+    type Output = O;
+    fn parse_state(&mut self, input: State<I>) -> ParseResult<O, I> {
+        if self.0.len() == 0 { // err: slice empty of parsers
+            return Err(Consumed::Empty(ParseError { position: input.position.clone(), errors: vec![] }))
+        }
+        for i in range(0, self.0.len()) {
+            match self.0[i].clone().parse_state(input.clone()) {
+                Err(_) => {},
```

Comment:
You actually need to distinguish between `Err(Consumed(_))` and `Err(Empty(_))`. Check the `or` parser to see how it deals with this.